monai/transforms/spatial/functional.py

```diff
@@ -229,7 +240,7 @@ def orientation(img, original_affine, spatial_ornt, lazy, transform_info) -> tor
     return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out  # type: ignore
 
 
-def flip(img, sp_axes, lazy, transform_info):
+def flip_image(img, sp_axes, lazy, transform_info):
```

Comment:
Similar to #7509, we could really just pass the appropriate world-space transform to flip, and limit the kind-specific code to applying an affine to the appropriate tensor type